oligo_designer_toolsuite/oligo_specificity_filter/_filter_bowtie.py

```diff
@@ -0,0 +1,337 @@
+import re
+import time
+from pathlib import Path
+from subprocess import Popen
+
+import pandas as pd
+from joblib import Parallel, delayed, parallel_backend
+
+from ._filter_base import ProbeFilterBase
+from ._utils import (
+    _load_probes_info,
+    _write_output,
+    _write_removed_genes,
+    mismatch_in_ligation,
+)
+
+
+class ProbeFilterBowtie(ProbeFilterBase):
+    def __init__(
+        self,
+        n_jobs,
+        dir_output,
+        dir_annotations,
+        file_transcriptome_fasta,
+        min_mismatches=4,
+        mismatch_region=None,
+        ligation_region=0,
+    ):
+        """This class filters probes based on the Bowtie short read alignment tool.
+        The user can customize the filtering by specifying the min_mismatches per probe and mismatch_region, the region that should be considered for counting mismatches.
+        That is, all probes with number mismatches less than min_mismatches inside the mismatch_region are filtered out.
+
+        Use conda install -c bioconda bowtie to install Bowtie package
+
+        :param file_transcriptome_fasta: path to fasta file containing all probes
+        :type file_transcriptome_fasta: str
+        :param min_mismatches: Threshhold value on the number of mismatches required for each probe. Probes where the number of mismatches are greater than or equal to this threshhold are considered valid. Possible values range from 0 to 4.
+        :type min_mismatches: int
+        :param mismatch_region: The region of the probe where the mismatches are considered. Probes that have less than min_mismatches in the first L bases (where L is 5 or greater) are filtered out
+        :type mismatch_region: int
+        :param ligation_region: Coverage between probes and target sequence should not span region around ligation site (e.g. ligation_region = 5 would correspond to -4 to +5 nt around ligation site). Probes with no mismatches in the ligation region are filtered out. If ligation_region = 0, omit this requirement
+        :type ligation_region: int
+
+
+        """
+        super().__init__(n_jobs, dir_output, dir_annotations)
+
+        self.file_transcriptome_fasta = file_transcriptome_fasta
+        self.ligation_region = ligation_region
+
+        if min_mismatches > 4:
+            raise ValueError(
+                "Choice of min_mismatches out of range for bowtie allignment tool. Please choose a value no greater than 4"
+            )
+        else:
+            self.min_mismatches = min_mismatches
+
+        if mismatch_region < 5:
+            raise ValueError(
+                "Choice of mismatch_region out of range for bowtie allignment tool. Please choose a value no less than 5"
+            )
+        else:
+            self.mismatch_region = mismatch_region
+
+        self.dir_bowtie = os.path.join(self.dir_output, "bowtie")
+        Path(self.dir_bowtie).mkdir(parents=True, exist_ok=True)
+
+        self.dir_probes = os.path.join(self.dir_output, "probes_bowtie")
```

Comment:
we should not write anything else than intermediate files, which are deleted afterwards, and store all other information in the probes DB